### PR TITLE
Add support for Google AutoML Translation

### PR DESCRIPTION
**Pull request by mathjazz**
This patch brings Google AutoML Translation support to Pontoon in addition to the existing Google Translation API support. [Read more about the differences between the two services.](https://cloud.google.com/translate/automl/docs/beginners-guide#is-the-translation-api-or-automl-the-right-tool-for-me)

AutoML Translation will be used instead of Translation API for locales for which we have trained models available. For that purpose, a new `Locale.google_automl_model` field has been added to the DB. If the value is set, Pontoon will use AutoML Translation with the given model ID, if not, it will use the Translation API with the generic model.

For now, we only have trained models available for Italian (it) and Slovenian (sl) locale, but we can easily add more. We chose these two locales, because we plan to use them to evaluate the Pretranslation service. In addition to that, AutoML Translation will be used in Machinery as well.

--

A couple of tweaks were required to make AutoML Translation work properly using the Google's client library:

1. An environment variable (`PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION`) is used to configure one of the libraries required by `google-cloud-translate`. See [039d903](https://github.com/mozilla/pontoon/pull/2648/commits/039d90335bb34e191b55e77d734b8623c1324c4c).

2. Authentication using [service accounts](https://cloud.google.com/translate/docs/setup?hl=en_US#auth) required adding a new buildpack to Heroku in order to bypass the limitations of the ephemeral storage. See [24a18c0](https://github.com/mozilla/pontoon/pull/2648/commits/24a18c0268a1d9d80968734f7465c8002af5dd19).

### CODE REVIEW

File: pontoon/machinery/utils.py [hunk 1]
```diff
@@ -61,6 +79,45 @@ def get_google_translate_data(text, locale_code):
...
+    try:
+        client = translate.TranslationServiceClient()
+    except DefaultCredentialsError as e:
+        log.error("Google AutoML Translation error: {e}")
+        return {
+            "status": False,
+            "message": f"{e}",
+        }
+
+    project_id = "85591518533"
```
Comment (eemeli) [inline pontoon/machinery/utils.py:92]: This magic constant should probably be a well commented config value, yes?

File: pontoon/machinery/utils.py [hunk 2]
```diff
@@ -61,6 +79,45 @@ def get_google_translate_data(text, locale_code):
...
+    model_id = locale.google_automl_model
+    location = "us-central1"
```
Comment (eemeli) [inline pontoon/machinery/utils.py:94]: This also looks like a magic constant. At the very least it should have a code comment explaining itself.

File: pontoon/machinery/utils.py [hunk 3]
```diff
@@ -61,6 +79,45 @@ def get_google_translate_data(text, locale_code):
...
+    parent = f"projects/{project_id}/locations/{location}"
+    model_path = f"{parent}/models/{model_id}"
+
+    response = client.translate_text(
+        request={
+            "contents": [text],
+            "target_language_code": locale.google_translate_code,
+            "model": model_path,
+            "source_language_code": "en",
```
Comment (eemeli) [inline pontoon/machinery/utils.py:103]: Just to check, is this always the case in Pontoon?

**Review by eemeli — CHANGES_REQUESTED** [review]
In general looks good; see inline comments regarding magic reduction.

File: pontoon/machinery/utils.py (hunk 3 above)
Comment (mathjazz) [reply pontoon/machinery/utils.py:103]: Yeah, and the same value is [hardcoded](https://github.com/mozilla/pontoon/blob/24a18c0268a1d9d80968734f7465c8002af5dd19/pontoon/machinery/utils.py#L51) in the generic API.

We track support for other source locales in https://github.com/mozilla/pontoon/issues/2052.

File: pontoon/machinery/utils.py (hunk 2 above)
Comment (mathjazz) [reply pontoon/machinery/utils.py:94]: I'll add a comment here - AutoML requires `us-central1`.

Although the docs are not [explicit](https://cloud.google.com/translate/automl/docs/before-you-begin#create_a_cloud_storage_bucket) about it, I tried other locations and got an error stating that.

File: pontoon/machinery/utils.py (hunk 1 above)
Comment (mathjazz) [reply pontoon/machinery/utils.py:92]: Of course, nice catch!

**Review by eemeli — APPROVED** [review]

**Pull request merged**